### PR TITLE
string solution to the palindrome number challenge

### PR DESCRIPTION
**Pull request by podmar**
Hi Oscar, here is the string version of the palindrome number challenge. If you request changes, I will try to add them and request a review again through the same PR :) Thank you!

### CODE REVIEW

File: palindrome_number.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+test_1 = 121
+test_2 = -121
+test_3 = 123
+
+def solution(x): 
+    is_palindrome = False
+
+    s = str(x)
+    s_backwards = s[-1::-1]
```
Comment (0sc) [inline palindrome_number.py]: With a negative step, the starting index doesn't matter as much. So the following might be simpler:
```python
s_backwards = s[::-1]
```

**Review by 0sc — CHANGES_REQUESTED** [review]
Looks good. Add a nitpick comment to help you practice reusing same PR.

File: palindrome_number.py (hunk 1 above)
Comment (podmar) [reply palindrome_number.py]: fixed 🙌🏻

**Comment by podmar** [comment]
I have implemented the change in the index and added another solution to the challenge (without changing to string). It broke my brain.

File: palindrome_number.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    if digit_list == backwards_digit_list: 
+        is_palindrome = True
+
+    return is_palindrome
+
+print(integer_solution(test_1), integer_solution(test_2), integer_solution(test_3), integer_solution(test_4))
+
+# Runtime: 82 ms, faster than 49.00% of Python online submissions for Palindrome Number.
+# Memory Usage: 13.4 MB, less than 66.69% of Python online submissions for Palindrome Number.
```
Comment (0sc) [inline palindrome_number.py:90]: Hmm ... this solution is pretty complex. Notice it's a bit slower than the first one. This is largely due to the extra cycle required to convert the number to an array and then reverse and compare the arrays.

An ideal integer (only) solution uses some math trick to reverse the integer. The trick here will be to use the [modulo operator](https://www.askpython.com/python/python-modulo-operator-math-fmod), `%`. _The gist of the operator is that it returns the remainder of a division operation._

Using modulo and some extra operators, we can have a reverse function like so:
```python
def reverse(num):
    reversed = 0
    while num > 0:
      reversed = (reversed * 10) + (num % 10)
      num = num // 10

    return reversed
```
(Try it: https://trinket.io/python/31bf271606)

Now reversed, you can complete the rest of the algorithm.

**Review by 0sc — CHANGES_REQUESTED** [review]

File: palindrome_number.py (hunk 2 above)
Comment (podmar) [reply palindrome_number.py:90]: Thank you! We did several exercises with modulo in my course, but somehow I totally forgot about it! Would you advice to replace the log10 with modulo entirely or just write another solution to the challenge?

Comment (0sc) [reply palindrome_number.py:90]: Sure, you can add it as another solution to the challenge

File: palindrome_number.py [hunk 3]
```diff
@@ -83,7 +84,29 @@ def integer_solution(x):
...
+            reversed = (reversed * 10) + (num % 10)
+            num = num // 10
+        return reversed
+
+    if reversed(x) == x: 
+        is_palindrome = True
+
+    return is_palindrome
+
+print(modulo_solution(test_1), modulo_solution(test_2), modulo_solution(test_3), modulo_solution(test_4))
```
Comment (0sc) [inline palindrome_number.py:109]: Eventually we should have these as real python tests :)

**Review by 0sc — APPROVED** [review]

**Comment by podmar** [comment]
thank you 👍 🙏🏻 they will be, but I would like to merge this branch first and then continue with the test in the other branch :)

**Pull request merged**